gcm/client.go

```diff
@@ -0,0 +1,103 @@
+package gcm
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// gcm Client const variables
+const (
+	DefaultGCMEndpoint = "https://fcm.googleapis.com/fcm/send"
+)
+
+// Client is GCM client
+type Client struct {
```

Comment:
Is there plan to make apns.Client and gcm.Client both to implementation of gunfish.Client interface?
I think, it seems good to dealing each Client in same interface.